File: tracking-program/8087/api/models/EquipmentData.js
/**
 * MeterData.js
 *
 * @description :: A set of data points for a power meter at a specific time.
 */

module.exports = {

  attributes: {

    // The power meter that reported the data.
    switch: {
      model: 'Switch',
    },

    // MAC address of the mesh note that reported the data.
    // Useful mainly for cases where the meter starts sending data before
    // being entered into the portal.
    meshId: {
      type: 'string'
    },

    // The time that the data was recorded by the meter.
    recordedAt: {
      type: 'number',
      required: true
    },

    // The day that the data was recorded, e.g. 2017-05-01.
    // Useful for doing aggregations.
    day: {
      type: 'string',
      required: true
    },

    // The minute of the hour that the data was recorded, 0 - 59.
    // Useful for doing aggregations.
    minute: {
      type: 'number',
      required: true
    },

    // The 15 minute interval during which this data was recorded, including time zone offset.
    // Example: '0/-6:00', '25/+2:00', '95/-5:00'.
    // Useful for doing aggregations.
    intervalId: {
      type: 'string',
      required: true
    },

    //  ┌─┐┌─┐┬ ┬┌─┐┬─┐  ┬─┐┌─┐┌─┐┌┬┐┬┌┐┌┌─┐┌─┐
    //  ├─┘│ ││││├┤ ├┬┘  ├┬┘├┤ ├─┤ │││││││ ┬└─┐
    //  ┴  └─┘└┴┘└─┘┴└─  ┴└─└─┘┴ ┴─┴┘┴┘└┘└─┘└─┘
    l1Volt: {
      type: 'number'
    },

    l1Amp: {
      type: 'number'
    },

    l1Kw: {
      type: 'number'
    },

    l1Kva: {
      type: 'number'
    },

    l1Pf: {
      type: 'number'
    },

    l1Kvar: {
      type: 'number'
    },

    l2Volt: {
      type: 'number'
    },

    l2Amp: {
      type: 'number'
    },

    l2Kw: {
      type: 'number'
    },

    l2Kva: {
      type: 'number'
    },

    l2Pf: {
      type: 'number'
    },

    l2Kvar: {
      type: 'number'
    },

    l3Volt: {
      type: 'number'
    },

    l3Amp: {
      type: 'number'
    },

    l3Kw: {
      type: 'number'
    },

    l3Kva: {
      type: 'number'
    },

    l3Pf: {
      type: 'number'
    },

    l3Kvar: {
      type: 'number'
    },

    totalVolt: {
      type: 'number'
    },

    totalAmp: {
      type: 'number'
    },

    totalKw: {
      type: 'number'
    },

    totalKva: {
      type: 'number'
    },

    totalPf: {
      type: 'number'
    },

    totalKvar: {
      type: 'number'
    },

    l1VoltTHD: {
      type: 'number'
    },
    l2VoltTHD: {
      type: 'number'
    },
    l3VoltTHD: {
      type: 'number'
    },
    l1AmpTHD: {
      type: 'number'
    },
    l2AmpTHD: {
      type: 'number'
    },
    l3AmpTHD: {
      type: 'number'
    },
    totalVoltTHD: {
      type: 'number'
    },
    totalAmpTHD: {
      type: 'number'
    },
  },

};

File: tracking-program/8087/flask_app/app/models/equipment_data.py
"""EquipmentData model - for scheduler/switch power readings. api/models/EquipmentData.js"""
from app.extensions import db
from app.models.base import BaseModel


class EquipmentData(BaseModel):
    __tablename__ = "equipmentdata"

    switch = db.Column(db.Integer, db.ForeignKey("switch.id"))
    meshId = db.Column(db.String(255))
    recordedAt = db.Column(db.Float, nullable=False)
    day = db.Column(db.String(255), nullable=False)
    minute = db.Column(db.Float, nullable=False)
    intervalId = db.Column(db.String(255), nullable=False)
    l1Volt = db.Column(db.Float)
    l1Amp = db.Column(db.Float)
    l1Kw = db.Column(db.Float)
    l1Kva = db.Column(db.Float)
    l1Pf = db.Column(db.Float)
    l1Kvar = db.Column(db.Float)
    l2Volt = db.Column(db.Float)
    l2Amp = db.Column(db.Float)
    l2Kw = db.Column(db.Float)
    l2Kva = db.Column(db.Float)
    l2Pf = db.Column(db.Float)
    l2Kvar = db.Column(db.Float)
    l3Volt = db.Column(db.Float)
    l3Amp = db.Column(db.Float)
    l3Kw = db.Column(db.Float)
    l3Kva = db.Column(db.Float)
    l3Pf = db.Column(db.Float)
    l3Kvar = db.Column(db.Float)
    totalVolt = db.Column(db.Float)
    totalAmp = db.Column(db.Float)
    totalKw = db.Column(db.Float)
    totalKva = db.Column(db.Float)
    totalPf = db.Column(db.Float)
    totalKvar = db.Column(db.Float)
